packages/odata-common/src/de-serializers/uri-value-converter.ts

```diff
@@ -14,6 +14,7 @@ export function isInfOrNan(value: string | number): boolean {
 }
 
 /**
+ * As per OData URL Syntax rules, single quotes within string literals must be represented as two consecutive single quotes.
```

Comment:
Perhaps add a second sentence that this is not considered by the `encodeURIComponent` on the high level `get-filter` and `get-resource-path`.